sktalk/corpus/conversation.py

```diff
@@ -68,3 +69,169 @@ def asdict(self):
+            kwargs (dict): information about the calculation to store in the Conversation metadata
+        """
+        if len(values) != len(self.utterances):
+            raise ValueError(
+                "The number of values must match the number of utterances")
+        metadata = {field: kwargs}
+        try:
+            self._metadata["Calculations"].update(metadata)
+        except KeyError:
+            self._metadata = {"Calculations": metadata}
+        for index, utterance in enumerate(self.utterances):
+            setattr(utterance, field, values[index])
+
+    def calculate_FTO(self, window: int = 10000, planning_buffer: int = 200, n_participants: int = 2):
+        """Calculate Floor Transfer Offset (FTO) per utterance
+
+        FTO is defined as the difference between the time that a turn starts and the
+        end of the most relevant prior turn by the other participant, which is not
+        necessarily the prior utterance.
+
+        An utterance does not receive an FTO if there are preceding utterances
+        within the window that do not have timing information, or if it lacks
+        timing information itself.
+
+        To be a relevant prior turn, the following conditions must be met, respective to utterance U:
+        - the utterance must be by another speaker than U
+        - the utterance by the other speaker must be the most recent utterance by that speaker
+        - the utterance must have started before utterance U, more than `planning_buffer` ms before.
+        - the utterance must be partly or entirely within the context window (`window` ms prior
+            to the start of utterance U)
+        - within the context window, there must be a maximum of `n_participants` speakers.
+
+        Args:
+            window (int, optional): the time in ms prior to utterance in which a
+                relevant preceding utterance can be found. Defaults to 10000.
+            planning_buffer (int, optional): minimum speaking time in ms to allow for a response.
+                Defaults to 200.
+            n_participants (int, optional): maximum number of participants overlapping with
+                the utterance and preceding window. Defaults to 2.
+        """
+        values = []
+        for index, utterance in enumerate(self.utterances):
+            sub = self._subconversation_by_time(
+                index=index,
+                before=window,
+                after=0,
+                exclude_utterance_overlap=True)
+            if not 2 <= sub.count_participants() <= n_participants:
+                values.append(None)
+                continue
+            potentials = [
+                u for u in sub.utterances if utterance.relevant_for_fto(u, planning_buffer)]
+            try:
+                relevant = potentials[-1]
+                values.append(relevant.until(utterance))
+            except IndexError:
+                values.append(None)
```

Comment:
To simplify the code and avoid having a try-except block in a loop:

```suggestion
            values.append(potentials[-1].until(utterance) if potentials else None)
```